[FIX] "Groups" lacks of buttons to add/remove users #443

chfsx opened the pull request:
This is a Fix for Issue #443, a ReviewApp will follow asap

chfsx committed:
[FIX] "Groups" lacks of buttons to add/remove users #443

chfsx committed:
[FIX] change owner and grant access fixed as well

chfsx committed:
[FIX] CSS changes in groups and invitations for ILIAS 9

chfsx committed:
[FIX] Button issues

chfsx committed:
[FIX] badge has no color

chfsx commented:
Includes a fix for #447  as well

chfsx committed:
[FIX] codestyle in JS

chfsx committed:
[FIX] prepared remove all

chfsx committed:
[FIX] issue with addAll button if there are none left

dagraf commented:
@ferishili: This is the PR to review (mentioned in https://github.com/opencast-ilias/OpenCast/issues/443#issuecomment-3153597848).

ferishili commented on templates/default/change_owner.css:
```diff
@@ -1,11 +1,40 @@
-    right: 5px;
-    padding: 5px 8px !important;
-}
+
```
Unwanted new line! Please remove it and keep it at max 1 new line so that it would look better.

ferishili commented on templates/default/change_owner.css:
```diff
@@ -1,11 +1,40 @@
-}
\ No newline at end of file
+}
+
```
Unwanted new line! Please remove it and keep it at max 1 new line so that it would look better.

ferishili commented on templates/default/groups.css:
```diff
@@ -1,11 +1,42 @@
+    background-color: #f63b4c;
+    border-color: #dc3545;
+}
+
```
Unwanted new line! Please remove it and keep it at max 1 new line so that it would look better.

ferishili commented on templates/default/invitations.js:
```diff
@@ -3,166 +3,191 @@
+  container: null,
+  filter_container: null,
+  filtering: false,
+  lng: [
```
I understand that it's written as array in all the other files as well, but it should actually be `object`, since `this.lng` is consistently accessed using dot notation throughout the code.
This could lead to issues if `lngFromJson` hasn't been called beforehand.
NOTE: This also applies to other places!

chfsx replied:
I changed that to objects in the files I made changes in this PR. Please note that those scripts are really old und a lot of this stuff should be refactored completely. there is a lot of duplicated stuff as well, but within the scope of these fixes i could not rebuild, centralise and improve this. see also my note regarding `.min`.

ferishili commented on templates/default/group_participants.min.js:
This file has been removed because we no longer use minified `.js` files, though that raises a question: why?

If we've officially decided to stop using `.min.js` files for this set of scripts, then we should consider removing the other minified files as well (e.g., `invitations.min.js`, `change_owner.min.js`, etc.) for consistency and cleanup.

On the other hand, if we are still supposed to support built/minified versions, then all related `.js` files should be built accordingly, and their paths and filenames should be updated in the corresponding PHP files.

chfsx replied:
as already mentioned, these scripts are very old, require a fundamental reorganisation and should, for example, be completely integrated into the build process of our JS files. from my point of view, however, this is a medium-sized undertaking, but one that we should certainly tackle at some point.

as part of this PR, i have not changed anything regarding the integration of the files, they were already not integrated in the .min version before. i have now removed the .min versions.

this doesn't mean that we can get away from them in general, but as I said, these files (change_owner.js, group_participant.js, groups.js and invitations.js) must be transferred to a clean JS/UI component.

ferishili reviewed (changes requested):
Thanks, @chfsx.
I've added my review with a few requests. The most important point is regarding whether we should remove or keep and build the `.min.js` files. I've addressed this specifically in my review comment on the file: `templates/default/group_participants.min.js`.

chfsx committed:
[FIX] Style Issues in Chat

commit f27a8aba7d8b9df2990f3ddd76d62d0a472cb33b
Author: FarbodZamani <53179227+ferishili@users.noreply.github.com>
Date:   Wed Aug 6 16:11:03 2025 +0200

    fix and upgrade opencast chat for ilias 9

commit 584190848a7bfa01fbc4ef70e0b10d55281ed8e3
Author: FarbodZamani <53179227+ferishili@users.noreply.github.com>
Date:   Wed Aug 6 16:10:11 2025 +0200

    upgrade node lib "mysql" to "mysql2"

chfsx committed:
[FEATURE] Upgrade paella-user-tracking to v1.42.6

commit 84bd1742ecacb7a307c3cda186cde786b4bc565d
Author: Sascha Nösberger <sascha.noesberger@unibe.ch>
Date:   Tue Jun 17 16:04:10 2025 +0200

    Roll-up build for paella-user-tracking v1.42.6

commit 7b577f2455c060c630e1098183ef5c1d7c8dd94f
Author: Sascha Nösberger <sascha.noesberger@unibe.ch>
Date:   Tue Jun 17 13:10:20 2025 +0200

    Upgrade paella-user-tracking to v1.42.6

    This version of the paella-user-tracking plugin contains the option to disable
    the use of beacons for Matomo. This option allows to improve of the user tracking
    with Matomo.

chfsx committed:
[FIX] changes due to review

chfsx commented:
@dagraf asked me to merge this PR asap after making the requested changes. This is now part of  v9.2.3

ferishili commented:
Were all the changes resolved? Is there no need for me to re-check the requested changes?
If so, please make sure I'm notified by simply clicking "Resolve conversation".

chfsx commented:
@ferishili I was asked to merge it directly after the changes, but the changes were all relatively small (newlines and array->object), and as mentioned a few deleted files (.min).

ferishili commented:
Oh, I see. I noticed the bunch of commits after the change request and thought it was going to be a huge change!
All good, thanks for the clarification.
